Typescript error.

node_modules/react-native-shimmer-placeholder/index.d.ts:19:5 - error TS1038: A 'declare' modifier cannot be used in an already ambient context.

19     declare class MyComponent extends React.Component<Props, any> {}
       ~~~~~~~


###############
react-native-shimmer-placeholder: 1.0.34
tsc version: 3.4.5

Remove declare seems fixed the issue.

--- a/index.d.ts
+++ b/index.d.ts
@@ -1,21 +1,23 @@
 declare module 'react-native-shimmer-placeholder' {
     import React from 'react';
-    type Props = {
-        width?: number | string,
-        height?: number | string,
-        widthShimmer?: number,
-        duration?: number,
-        delay?: number,
-        colorShimmer?: string[],
-        reverse?: boolean,
-        autoRun?: boolean,
-        visible?: boolean,
-        children?: any,
-        style?: any,
-        backgroundColorBehindBorder?: string,
-        hasBorder?: boolean,
-        isInteraction?: boolean,
+    import { Animated } from 'react-native';
+    interface Props {
+        width?: number | string;
+        height?: number | string;
+        widthShimmer?: number;
+        duration?: number;
+        delay?: number;
+        colorShimmer?: string[];
+        reverse?: boolean;
+        autoRun?: boolean;
+        visible?: boolean;
+        children?: any;
+        style?: any;
+        backgroundColorBehindBorder?: string;
+        hasBorder?: boolean;
+        isInteraction?: boolean;
     }
-    declare class MyComponent extends React.Component<Props, any> {}
-    export default MyComponent
+    export default class ShimmerPlaceHolder extends React.Component<Props, any> {
+        getAnimated(): Animated.CompositeAnimation;
+    }
 }

